app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java

```diff
@@ -672,6 +686,11 @@ public static Intent getIntentByLink(final Context context,
                     + " url=" + url);
         }
 
+        if (linkType == StreamingService.LinkType.SEARCH) {
+            return getOpenSearchIntent(context, url, service.getServiceId(),
+                    linkType, service.getIdByUrl(url));
```

Comment:
Intead of obtaining the ID this way, as said in https://github.com/TeamNewPipe/NewPipeExtractor/pull/950#discussion_r1006825163, I think you should use `service.getSearchLHFactory().getId(url)`.